From 59210ad3c8d0ac75107d5e82f1666f67d7647ba6 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Wed, 5 Jul 2017 15:44:11 +0800
Subject: [PATCH 2053/5242] MLK-15342-1 soc: imx: i.mx8mq uses its own
 cpu-freq driver

commit  6694378fe74cae886d9e20a644f946576af2794c from
https://source.codeaurora.org/external/imx/linux-imx.git

i.MX8MQ is a SMP SoC without system controller, so
it will has its own cpu-freq driver, add machine
check for different platforms.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/soc-imx8.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/imx/soc-imx8.c b/drivers/soc/imx/soc-imx8.c
index 19ec917..26823c3 100644
--- a/drivers/soc/imx/soc-imx8.c
+++ b/drivers/soc/imx/soc-imx8.c
@@ -149,7 +149,10 @@ static int __init imx8_soc_init(void)
 
 static int __init imx8_register_cpufreq(void)
 {
-	platform_device_register_simple("imx8-cpufreq", -1, NULL, 0);
+	if (of_machine_is_compatible("fsl,imx8mq"))
+		platform_device_register_simple("imx8mq-cpufreq", -1, NULL, 0);
+	else
+		platform_device_register_simple("imx8-cpufreq", -1, NULL, 0);
 
 	return 0;
 }
-- 
1.7.9.5

